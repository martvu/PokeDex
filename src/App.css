#root {
  font-family: "Flexo-Demi", Arial, Helvetica, sans-serif;
  height: 100%;
  --bg-color: rgb(255, 255, 255); 
  --card-color: rgba(146, 149, 158, 0.73);
  --primary-color:#ba5050;
  --secondary-color: rgb(255, 255, 255);
  --text-color: #333;
  --white-color: rgb(255, 255, 255, 0.87);
  --accent-color: #ffcb05;
  --gradient-one: rgb(255, 255, 255);
  --gradient-two:rgb(224, 223, 223);
  --header-text: #ffcb05;
  display: flex;
  justify-content: center;
  align-items: center;
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  line-height: 1.5;
}

html, body {
  height: 100%;
  margin: 0px;
  background-color: var(--bg-color);
}

html {
  font-size: 16px;
}

body {
  overflow: auto;
  height: auto;
  color: var(--text-color);
  background-color: var(--bg-color);
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
} 

::-webkit-scrollbar {
  width: 0px;
  background: transparent;
  height: 0px;
}

.app {
  height: 100%;
  background-color: var(--bg-color);
}

/* Loading screen */
.loading-screen {
  margin-top: 2rem;
}

.loading-screen img{
  width: 300px;
  filter: invert(0.7);
}

/* Header */
.header {
  width: 100%;
  height: 80px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: sticky; /* Make it stick to the top */
  top: 0; /* Stick to the top */
  left: 0; /* Stick to the left */
  right: 0; /* Stick to the right */
  z-index: 102; /* Stay on top of everything */
  background-color: var(--secondary-color);
  overflow:hidden;
  box-shadow: 0 0 10px 1px rgb(0, 0, 0, 0.5);
}

.header img {
  width: 250px;
  z-index: -1;
  position: absolute;
  top: -95px;
  left: -100px;
  opacity: 0.8;
  filter: invert(0.7);
}

.header h1 {
  margin: 0.5rem 0;
  color: var(--header-text);

  text-shadow: rgb(61, 125, 202) 2px 0px 0px,
    rgb(61, 125, 202) 1.75517px 0.958851px 0px,
    rgb(61, 125, 202) 1.0806px 1.68294px 0px,
    rgb(61, 125, 202) 0.141474px 1.99499px 0px,
    rgb(61, 125, 202) -0.832294px 1.81859px 0px,
    rgb(61, 125, 202) -1.60229px 1.19694px 0px,
    rgb(61, 125, 202) -1.97998px 0.28224px 0px,
    rgb(61, 125, 202) -1.87291px -0.701566px 0px,
    rgb(61, 125, 202) -1.30729px -1.5136px 0px,
    rgb(61, 125, 202) -0.421592px -1.95506px 0px,
    rgb(61, 125, 202) 0.567324px -1.91785px 0px,
    rgb(61, 125, 202) 1.41734px -1.41108px 0px,
    rgb(61, 125, 202) 1.92034px -0.558831px 0px, rgb(0, 58, 112) 7px 0px 0px,
    rgb(0, 58, 112) 6.92869px 0.996602px 0px,
    rgb(0, 58, 112) 6.71622px 1.9729px 0px,
    rgb(0, 58, 112) 6.36692px 2.909px 0px,
    rgb(0, 58, 112) 5.8879px 3.78584px 0px,
    rgb(0, 58, 112) 5.28893px 4.58555px 0px,
    rgb(0, 58, 112) 4.5822px 5.29183px 0px,
    rgb(0, 58, 112) 3.78212px 5.8903px 0px,
    rgb(0, 58, 112) 2.90498px 6.36876px 0px,
    rgb(0, 58, 112) 1.96865px 6.71747px 0px,
    rgb(0, 58, 112) 0.992221px 6.92932px 0px,
    rgb(0, 58, 112) -0.00442571px 7px 0px,
    rgb(0, 58, 112) -1.00098px 6.92806px 0px,
    rgb(0, 58, 112) -1.97715px 6.71498px 0px,
    rgb(0, 58, 112) -2.91303px 6.36508px 0px,
    rgb(0, 58, 112) -3.78956px 5.88551px 0px,
    rgb(0, 58, 112) -4.58889px 5.28603px 0px,
    rgb(0, 58, 112) -5.29472px 4.57885px 0px,
    rgb(0, 58, 112) -5.89269px 3.77839px 0px,
    rgb(0, 58, 112) -6.3706px 2.90095px 0px,
    rgb(0, 58, 112) -6.71871px 1.96441px 0px,
    rgb(0, 58, 112) -6.92995px 0.98784px 0px,
    rgb(0, 58, 112) -6.99999px -0.00885142px 0px,
    rgb(0, 58, 112) -6.92743px -1.00536px 0px,
    rgb(0, 58, 112) -6.71372px -1.98139px 0px,
    rgb(0, 58, 112) -6.36324px -2.91705px 0px,
    rgb(0, 58, 112) -5.88311px -3.79328px 0px,
    rgb(0, 58, 112) -5.28313px -4.59223px 0px,
    rgb(0, 58, 112) -4.57551px -5.29762px 0px,
    rgb(0, 58, 112) -3.77466px -5.89507px 0px,
    rgb(0, 58, 112) -2.89692px -6.37243px 0px,
    rgb(0, 58, 112) -1.96016px -6.71995px 0px,
    rgb(0, 58, 112) -0.983458px -6.93057px 0px,
    rgb(0, 58, 112) 0.0132771px -6.99999px 0px,
    rgb(0, 58, 112) 1.00974px -6.92679px 0px,
    rgb(0, 58, 112) 1.98564px -6.71247px 0px,
    rgb(0, 58, 112) 2.92107px -6.36139px 0px,
    rgb(0, 58, 112) 3.797px -5.88071px 0px,
    rgb(0, 58, 112) 4.59557px -5.28022px 0px,
    rgb(0, 58, 112) 5.30051px -4.57215px 0px,
    rgb(0, 58, 112) 5.89746px -3.77094px 0px,
    rgb(0, 58, 112) 6.37426px -2.89289px 0px,
    rgb(0, 58, 112) 6.72119px -1.95591px 0px,
    rgb(0, 58, 112) 6.93119px -0.979076px 0px;
}

.header-link-container {
  text-decoration: none;
  color: var(--text-color);
  font-size: 1.5rem;
  font-weight: 600;
  transition: transform 0.2s ease-in-out;
}

.header-link-container:hover {
  opacity: 0.8;
}

.link {
  text-decoration: none;
}

/* Filters */
.filter-container {
  padding: 5px 0;;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  width: 100%;
  background-color: var(--primary-color);
  position: sticky;
  top: 80px;
  z-index: 103;
  box-shadow:0px 5px 10px 0.2px rgb(0, 0, 0, 0.5) ;
}

.filter-reset-btn {
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  border: none;
  border-radius: 1rem;
  padding: 0.5rem;
  background-color: transparent;
}

.filter-reset-btn:active {
  transform: scale(0.9);
}
.search-bar-container {
  display: flex;
  align-items: center;
  position: relative;
}

.search-icon {
  position: relative;
  left: 25px;
  scale: 1.5;
  color: var(--text-color);
}

.search-bar {
  border-radius: 1rem;
  padding: 0 1rem 0 2rem;
  height: 2rem;
  width: 12em;
  font-size: 1.1rem;
  font-weight: 400;
  border: none;
  color: var(--text-color);
  background-color: var(--secondary-color);
}

.filter-options-container {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0.5rem;
}

.filter-favorite-btn {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 0.5rem;
  margin: 0 0.5rem 0 1rem;
  padding: 0.5rem 0.8rem;
  background-color: var(--secondary-color);
  cursor: pointer;
  transition: transform 0.2s ease-in-out;
}

.filter-favorite-btn.show {
  background-color: var(--secondary-color);
  box-shadow: 0 0 5px 4px rgba(255, 255, 255, 0.2);
}

.filter-favorite-btn:active {
  transform: scale(0.9);
}

.filter-favorite-btn:hover {
  transform: scale(1.1);
}

.filter-select {
  margin: 0 0.5rem;
}
.filter-select label {
  color: var(--text-color);
}

.filter-select select {
  border: none;
  border-radius: 0.5rem;
  padding: 0.3rem 0.5rem;
  margin-left: 0.5rem;
  font-size: 1rem;
  font-weight: 600;
  background-color: var(--secondary-color);
  color: var(--text-color);
  cursor: pointer;
  transition: transform 0.2s ease-in-out;
}

/* PokemonList */
.bg-container {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  max-width: 1500px;
  height: auto;
  display: flex; 
  justify-content: center;
  align-items: center;
  flex-direction: column;
  flex-wrap: wrap;
  background-color: transparent;
}

.pokemon-list-container {
  list-style-type: none;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  flex-wrap: wrap;
  padding: 0.5rem 0.5rem 0 0.5rem;
  margin: 0;
  animation: fadeIn 1s ease-in-out;
  max-width: 1000px;
  /* box-shadow: 0 5px 10px 2px rgb(0, 0, 0, 0.5); */
}

.pokemon-list-container:empty {
  display: none;
}

/*Pokemon Card*/
.pokemon-card-container {
  /* border: 1px solid var(--secondary-color);  */
  border-radius: 1rem;
  display: flex;
  align-items: center;
  min-width: 170px;
  max-width: 170px;
  height: 180px;
  max-height: 180px;
  margin: 0.8rem;
  transition: all 0.3s ease-in-out;
  box-shadow: 0 5px 20px 2px rgb(0 0 0 / 50%);
  cursor: pointer; 
  position: relative;
  overflow: hidden;
  background-color: var(--card-color);
}

.pokemon-name {
  font-family: "Flexo-Demi", Arial, Helvetica, sans-serif;
  font-weight: 700;
  position: absolute;
  left: 10px;
  top: 10px;
  font-size: 1.2rem;
  margin-bottom: 0.5rem;
  color: var(--white-color);
}
.favorite-button {
  position: absolute;
  width: 20px;
  height: 20px;
  top: 15px;
  right: 10px;
  scale: 1.4;
  background-color: transparent;
  border: none;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  z-index: 5;
}
.favorite-button:hover {
  scale: 1.6;
  outline: none;
}
.favorite-button:focus {
  outline: none;
}
.favorite-icon-inactive {
  color: var(--white-color);
}
.favorite-icon-active {
  color: red;
  animation: heart-beat 0.5s; 
}

.pokemon-id {
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  position: absolute;
  bottom: 10px;
  left: 10px;
  opacity: 0.8;
  font-size: 1.5rem;
  font-weight: 600;
  color: var(--white-color)
}

.type-container {
  position: absolute;
  top: 45px;
  left: 10px;
}

.type-icon {
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  display: flex;
  justify-content: left;
  align-items: center;
  border-radius: 0.5rem;
  font-weight: 800;
  font-size: 0.7rem;
  padding: 0.2rem 0.5rem;
  margin-bottom: 0.4rem;
  box-shadow: 0 0 10px 1px rgb(0, 0, 0, 0.5);
  color: var(--white-color)
}

.type-icon img {
  width: 10px;
  height: 10px;
  margin-right: 3px;
}

.sprite-bg-container {
  position: absolute;
  width: 120px;
  opacity: 0.1;
  top: 80px;
  right: -10px;
  filter: invert(1);
}

.sprite-bg {
  width: 100%;
  height: 100%;
  opacity: 0.8;
}

.sprite-container {
  width: 80px;
  height: 80px;
  z-index: 1;
  position: absolute;
  top: 80px;
  right: 10px;
}

.pokemon-sprite {
  width: 100%;
  height: 100%;
  transition: all 0.2s ease-in-out;
}

.pokemon-sprite:hover {
  scale: 1.1;
}
/* PokemonInfoPage */
.info-page-container {
  max-width: 700px;
  margin: 1rem 0;
  color: white;
  display:flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
} 

.info-bg-container h5 {
  font-size: 1.1rem;
  margin: 5px;
}

.filler-div {
  position: sticky;
  top: 80px;
  width: 100%;
  height: 30px;
  z-index: 103;
  background-color: var(--primary-color);
  box-shadow: 0 5px 10px 2px rgb(0 0 0 / 50%);
}
.info-pokeball-icon {
  position: absolute;
  width: 250px;
  opacity: 0.4;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  z-index: -1;
}

.info-pokemon-card {
  width: 100%;
  max-width: 500px;
  margin: 1rem auto;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  border-radius: 0.5rem;
  border: solid 1px #a7a7a7;
  box-shadow: 0 0 10px 1px rgb(0, 0, 0, 0.5);
  padding: 1rem;
  /* background: linear-gradient(to bottom, rgba(60, 60, 60, 0.5) 0%, rgba(0, 0, 0, 0.5) 100%); */
}

.info-pokemon-card h1, h4 {
  text-align: center;
  text-transform: capitalize;
}
@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.info-pokemon-sprite {
  width: 200px;
  height: 200px;
  margin: 0 auto;
  animation: fadeIn 1s ease-in-out;
}

.info-type-list {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  margin: 0.5rem 0;
}

.info-type-badge {
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  display: flex;
  justify-content: left;
  align-items: center;
  border-radius: 0.5rem;
  font-weight: 800;
  font-size: 0.7rem;
  padding: 0.2rem 0.5rem;
  margin: 0.2rem;
  box-shadow: 0 0 10px 1px rgb(0, 0, 0, 0.5);
  color: var(--text-color)
}

.info-type-badge img {
  width: 10px;
  height: 10px;
  margin-right: 3px;
}

.pokemon-description {
  width: 100%;
  text-align: left;
  margin: 0.5rem;
  max-width: 500px;
  border: solid 1px #a7a7a7;
  padding: 0.3rem 0.5rem;
  border-radius: 0.5rem;
  box-shadow: 0 0 10px 1px rgb(0, 0, 0, 0.5);
  color: var(--text-color);
  background: linear-gradient(to bottom, var(--gradient-one) 0%, var(--gradient-two) 100%);
}

.pokemon-description h5 {
  margin: 0.5rem 0;
}

.next-btn, .prev-btn {
  background-color: var(--accent-color); /* Darken the background color */
  border: solid 2px rgb(0, 58, 112);
  border-radius: 10px;
  padding: 2px 5px;
  color: rgb(0, 58, 112);
  font-size: 1.3rem;
  font-weight: 600;
  cursor: pointer;
  margin: 0 0.5rem;
  transition: background-color 0.3s, color 0.3s;
  transform: scale(1.1);
}

.next-btn:hover, .prev-btn:hover {
   opacity: 0.8;
}

.pokemon-dimensions {
  display: flex;
  width: 100%;
  justify-content: space-evenly;
  margin: 10px 0;
}

.pokemon-height, .pokemon-weight {
  border: solid 1px #a7a7a7;
  padding: 0.5rem;
  border-radius: 0.5rem;
  box-shadow: 0 0 10px 1px rgb(0, 0, 0, 0.5);
  color: var(--text-color);
  background: linear-gradient(to bottom, var(--gradient-one) 0%, var(--gradient-two) 100%);
}

.pokemon-stats {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  height: 100%;
  padding: 1em 0 3em 0;
  border-radius: 0.5em;
  border: solid 1px #a7a7a7;
  box-shadow: 0 0 10px 1px rgb(0, 0, 0, 0.5);
  color: var(--text-color);
  background: linear-gradient(to bottom, var(--gradient-one) 0%, var(--gradient-two) 100%);
}

.pokemon-stats h5 {
  text-align: right;
  text-transform: capitalize;
}

.parameter-container {
  width: 80%;
}

.statbar-container {
  height: 18px;
  width: 100%;
  background-color: #a7a7a7;
  border-radius: 1rem;
  margin-top: 3px;
}

.statbar-segment {
  height: 18px;
  display: inline-block;
  border-radius: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: 700;
  font-size: 0.8rem;
  min-width: 5%;
  animation: statSlideIn 2s ease-in-out;
  filter: hue-rotate(15deg);
}
  
@media screen and (max-width: 768px) {
  #root {
    padding: 0rem;
  }

  .header {
    height: 60px;
  }
  
  .filler-div {
    top: 60px;
  }
  .theme-switch {
    top: 5px;
    right: 10px;
    scale: 0.8;
  }
  .filter-container {
    top: 60px;
  }
  .filter-options-container {
    margin: 0.5rem 0;
  }
  .search-bar {
    font-size: 1rem;
  }

}

@media screen and (max-width: 500px) {
  .pokemon-card, 
  .info-pokemon-card, 
  .pokemon-description,
  .pokemon-stats,
  .pokemon-card-info{
    width: 90%;
  }
}

@media screen and (max-width: 450px) {
  #root {
    padding: 0rem;
  }
  .pokemon-list-container {
    margin: 0;
    padding: 0.2rem 0;
  }

  .filter-container {
    padding: 0.5rem 0 0 0;
  }
  .pokemon-card-container {
    margin: 0.6rem 0.2rem 0.3rem 0.2rem;
    scale: 0.95;
  }
}
