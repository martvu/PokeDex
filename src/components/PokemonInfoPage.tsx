import { getPokemonData } from '../utils/pokeApi';
import { useQuery } from '@tanstack/react-query';
import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { TYPE_COLORS, STAT_COLORS } from '../utils/constants';
import pokeBall from '../assets/img/pb-icon.svg';
import { FlavorText, PokemonStat, PokemonType } from '../utils/pokeApiTypes';
import { formatPokemonName, getTypeColorGradient } from '../utils/utils';

function PokemonInfoPage() {
  const [id, setId] = useState(1);
  const maxNumPokemon = 151;
  const [cardColor, setCardColor] = useState({});
  const url = `https://pokeapi.co/api/v2/pokemon/${id}/`;
  const {
    data: pokemonDetails,
    isLoading,
    error,
  } = useQuery(['pokemon', url], () => getPokemonData(url));
  const speciesUrl = `https://pokeapi.co/api/v2/pokemon-species/${id}/`;
  const { data: speciesData, isLoading: isLoadingSpecies } = useQuery(
    ['species', speciesUrl],
    () => getPokemonData(speciesUrl)
  );

  console.log(pokemonDetails);
  console.log(speciesData);
  useEffect(() => {
    if (!pokemonDetails) {
      console.log("no pokemon details");
      return
    };
    const typeColorGradient = getTypeColorGradient(pokemonDetails.types);
    setCardColor({
      background: `linear-gradient(to bottom, ${typeColorGradient[0]} 10%, ${typeColorGradient[1]} 100%`,
    });
  }, [pokemonDetails]);

  if (error) {
    return <div>Something went wrong...</div>;
  }
  return (
    <>
    {!isLoading && !isLoadingSpecies && pokemonDetails && speciesData && (
      <div className="info-bg-container">
        <div className="info-home-btn">
          <Link to="/">
            <button>Home</button>
          </Link>
        </div>

        {/* <input
      type="number"
      value={id}
      min={1}
      max={maxNumPokemon}
      onChange={e => setId(parseInt(e.target.value))}
    />  */}
        <div className="pokemon-card info" style={cardColor}>
          <div className="pokemon-name-id">
            <h1>{formatPokemonName(pokemonDetails?.name)}</h1>
            <h4 className="pokemon-text">
              {'#' + ('00' + pokemonDetails?.id).slice(-3)}
            </h4>
          </div>
          <div className="info-pokemon-img">
            <img className="pokeball-icon" src={pokeBall} alt="pokeball icon" />
            <img
              className="pokemon-sprite"
              src={
                pokemonDetails?.sprites?.other?.['official-artwork']
                  ?.front_default || '#'
              }
              alt={pokemonDetails.name}
            />
          </div>
          <div className="info-pokemon-data">
            <div className="type-list">
              {pokemonDetails.types.map((type: PokemonType) => (
                <span
                  key={type.slot}
                  className="type-badge"
                  style={{ backgroundColor: `${TYPE_COLORS[type.type.name]}` }}
                >
                  {type.type.name.charAt(0).toUpperCase() +
                    type.type.name.slice(1)}
                </span>
              ))}
            </div>
            <div className="pokemon-genus">
              {isLoadingSpecies ? 'Loading...' : speciesData?.genera[7]?.genus}
            </div>
          </div>
        </div>
        <div>
          <button 
            onClick={() => setId(id - 1)} 
            disabled={id === 1}
            className="prev-btn">
            Prev
          </button>
          <button 
            onClick={() => setId(id + 1)} 
            disabled={id === maxNumPokemon}
            className="next-btn">
            Next
          </button>
        </div>
        <div className="pokemon-description">
          <h5 className="pokemon-text info-text">Description</h5>
          {isLoadingSpecies ? (
            <p>Loading...</p>
          ) : (
            <p>
              {
                speciesData?.flavor_text_entries
                  .slice()
                  .reverse()
                  .find((flavor: FlavorText) => flavor?.language?.name === 'en')
                  ?.flavor_text
              }
            </p>
          )}
        </div>
        <div className="pokemon-dimensions">
          <div className="pokemon-height">
            <h5>Height</h5>
            <span>{pokemonDetails.height / 10}m</span>
          </div>
          <div className="pokemon-weight">
            <h5>Weight</h5>
            <span>{pokemonDetails.weight / 10}kg</span>
          </div>
        </div>
        <div className="pokemon-stats">
          <h5 className="pokemon-text info-text">Stats</h5>
          <div className="parameter-container">
            {pokemonDetails?.stats.map((stat: PokemonStat) => {
              return (
                <div key={stat.stat.name} className="parameter-section">
                  <h5 className="info-text">{stat.stat.name}</h5>
                  <div className="statbar-container">
                    <div
                      className="statbar-segment"
                      style={{
                        backgroundColor: STAT_COLORS[stat.stat.name],
                        width: `${(stat.base_stat / 255) * 100}%`,
                      }}
                    >
                      <span>{stat.base_stat}</span>
                    </div>
                  </div>
                </div>
              );
            })}
          </div>
        </div>
        <div className="pokemon-gender">
          <h5>Gender Ratio</h5>
          <div className="gender-ratio-container">
            {isLoadingSpecies ? (
              <span>Loading...</span>
            ) : speciesData.gender_rate === -1 ? (
              <span>Gender Unknown</span>
            ) : (
              <>
                <div
                  className="gender-ratio-segment"
                  style={{
                    backgroundColor: '#3355FF',
                    width: `${100 - speciesData.gender_rate * 12.5}%`,
                    borderRadius:
                      speciesData.gender_rate === 0 ? '1rem' : '1rem 0 0 1rem',
                  }}
                ></div>
                <div
                  className="gender-ratio-segment"
                  style={{
                    backgroundColor: '#FF77DD',
                    width: `${speciesData.gender_rate * 12.5}%`,
                    borderRadius:
                      speciesData.gender_rate === 8 ? '1rem' : '0 1rem 1rem 0',
                  }}
                ></div>
              </>
            )}
          </div>
          <div
            className="gender-percentages"
            style={{
              opacity: isLoadingSpecies
                ? 0
                : speciesData.gender_rate === -1
                ? 0
                : 1,
            }}
          >
            <span style={{ color: '#6982ff' }}>
              {isLoadingSpecies ? '-' : 100 - speciesData.gender_rate * 12.5}%
              male,{' '}
            </span>
            <span style={{ color: '#FF77DD' }}>
              {isLoadingSpecies ? '-' : speciesData.gender_rate * 12.5}% female
            </span>
          </div>
        </div>
      </div>
      )}
    </>
  );
}

export default PokemonInfoPage;
