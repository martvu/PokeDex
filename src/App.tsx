import './App.css';
import Pokemoncard from './components/pokemoncard';
import { Pokemon, NamedAPIResource, PokemonAbility, PokemonSprites, PokemonType, PokemonStat } from './utils/pokeApiTypes';  // Replace with the actual path to your types file


const pikachu: Pokemon = {
  id: 25, 
  name: "Pikachu", // Replace with the desired name
  base_experience: 112, // Replace with the desired base experience
  height: 4, // Replace with the desired height
  is_default: true, // Replace with the desired value
  order: 25, // Replace with the desired order
  weight: 60, // Replace with the desired weight
  abilities: [
    {
      is_hidden: false, // Replace with the desired value
      slot: 1, // Replace with the desired slot
      ability: {
        name: "Static", // Replace with the ability name
        url: "https://pokeapi.co/api/v2/ability/1/", // Replace with the actual URL
      },
    },
  ] as PokemonAbility[],

  forms: [] as NamedAPIResource[], // Replace with the desired forms

  game_indices: [] as any[], // Replace with the desired game indices

  held_items: [] as any[], // Replace with the desired held items

  location_area_encounters: "", // Replace with the desired location area encounters

  moves: [] as any[], // Replace with the desired moves

  sprites: {
    front_default: "https://example.com/pikachu-front.png", // Replace with the desired sprite URLs
    front_shiny: "https://example.com/pikachu-front-shiny.png", // Replace with the desired sprite URLs
    front_female: "https://example.com/pikachu-front-female.png", // Replace with the desired sprite URLs
    front_shiny_female: "https://example.com/pikachu-front-shiny-female.png", // Replace with the desired sprite URLs
    back_default: "https://example.com/pikachu-back.png", // Replace with the desired sprite URLs
    back_shiny: "https://example.com/pikachu-back-shiny.png", // Replace with the desired sprite URLs
    back_female: "https://example.com/pikachu-back-female.png", // Replace with the desired sprite URLs
    back_shiny_female: "https://example.com/pikachu-back-shiny-female.png", // Replace with the desired sprite URLs
  } as PokemonSprites,

  species: {
    name: "pikachu-species", // Replace with the species name
    url: "https://pokeapi.co/api/v2/pokemon-species/25/", // Replace with the actual URL
  } as NamedAPIResource,

  stats: [
    {
      stat: {
        name: "hp", // Replace with the stat name
        url: "https://pokeapi.co/api/v2/stat/1/", // Replace with the actual URL
      },
      effort: 0, // Replace with the effort value
      base_stat: 35, // Replace with the base stat value
    } as PokemonStat,
  ],

  types: [
    {
      slot: 1, // Replace with the desired slot
      type: {
        name: "electric", // Replace with the type name
        url: "https://pokeapi.co/api/v2/type/13/", // Replace with the actual URL
      },
    } as PokemonType,
  ],
};

function App() {
  return (
    <div>
      <div> Hello</div>
      <Pokemoncard pokemonInfo={pikachu}></Pokemoncard>
    </div>
  )
}

export default App;
