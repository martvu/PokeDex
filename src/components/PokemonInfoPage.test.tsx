import { render, renderHook, screen, waitFor } from '@testing-library/react';
import { describe, test, expect, vi } from 'vitest';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import App from '../App';
import PokemonInfoPage from './PokemonInfoPage';

const queryClient = new QueryClient();

global.fetch = vi.fn().mockResolvedValue({
  json: () =>
    Promise.resolve({
      id: 1,
      name: 'bulbasaur',
      height: 7,
      weight: 69,
    }),
}); 

describe('PokemonInfoPage', () => {
  test('renders PokemonInfoPage component', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <PokemonInfoPage />
      </QueryClientProvider>
    );
  });
});
