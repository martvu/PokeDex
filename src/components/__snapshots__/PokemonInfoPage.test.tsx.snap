// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot > snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="info-bg-container"
    >
      <div
        class="info-home-btn"
      >
        <a
          href="/"
        >
          <button>
            Home
          </button>
        </a>
      </div>
      <div
        class="info-pokemon-card"
      >
        <div
          class="pokemon-name-id"
        >
          <h1>
            Bulbasaur
          </h1>
          <h4
            class="pokemon-text"
          >
            #001
          </h4>
        </div>
        <div
          class="info-pokemon-img"
        >
          <img
            alt="bulbasaur"
            class="info-pokemon-sprite"
            src="#"
          />
        </div>
        <div
          class="info-pokemon-data"
        >
          <div
            class="info-type-list"
          >
            <span
              class="info-type-badge"
              style="background-color: rgb(122, 199, 76);"
            >
              Grass
            </span>
            <span
              class="info-type-badge"
              style="background-color: rgb(163, 62, 161);"
            >
              Poison
            </span>
          </div>
          <div
            class="pokemon-genus"
          >
            Seed Pokémon
          </div>
        </div>
      </div>
      <div>
        <button
          class="prev-btn"
          disabled=""
        >
          Prev
        </button>
        <button
          class="next-btn"
        >
          Next
        </button>
      </div>
      <div
        class="pokemon-description"
      >
        <h5
          class="pokemon-text info-text"
        >
          Description
        </h5>
        <p>
          Bulbasaur can be seen napping in bright sunlight.
There is a seed on its back. By soaking up the sun’s
rays, the seed grows progressively larger.
        </p>
      </div>
      <div
        class="pokemon-dimensions"
      >
        <div
          class="pokemon-height"
        >
          <h5>
            Height
          </h5>
          <span>
            NaNm
          </span>
        </div>
        <div
          class="pokemon-weight"
        >
          <h5>
            Weight
          </h5>
          <span>
            NaNkg
          </span>
        </div>
      </div>
      <div
        class="pokemon-stats"
      >
        <h5
          class="pokemon-text info-text"
        >
          Stats
        </h5>
        <div
          class="parameter-container"
        >
          <div
            class="parameter-section"
          >
            <h5
              class="info-text"
            >
              hp
            </h5>
            <div
              class="statbar-container"
            >
              <div
                class="statbar-segment"
                style="background-color: rgb(255, 1, 0); width: 17.647058823529413%;"
              >
                <span>
                  45
                </span>
              </div>
            </div>
          </div>
          <div
            class="parameter-section"
          >
            <h5
              class="info-text"
            >
              attack
            </h5>
            <div
              class="statbar-container"
            >
              <div
                class="statbar-segment"
                style="background-color: rgb(240, 128, 48); width: 19.215686274509807%;"
              >
                <span>
                  49
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
